glib/src/subclass/signal.rs

```diff
@@ -68,9 +68,103 @@ impl fmt::Debug for SignalInvocationHint {
     }
 }
 
+/// In-depth information of a specific signal
+#[repr(transparent)]
+pub struct SignalQuery(gobject_ffi::GSignalQuery);
+
+impl SignalQuery {
+    pub fn signal_name(&self) -> crate::GString {
+        unsafe { from_glib_none(self.0.signal_name) }
+    }
+
+    pub fn signal_id(&self) -> SignalId {
+        unsafe { SignalId::new(from_glib(self.0.itype), self.0.signal_id) }
+    }
+
+    pub fn type_(&self) -> Type {
+        unsafe { from_glib(self.0.itype) }
+    }
+
+    pub fn flags(&self) -> SignalFlags {
+        unsafe { from_glib(self.0.signal_flags) }
+    }
+
+    pub fn return_type(&self) -> Type {
+        unsafe { from_glib(self.0.return_type) }
+    }
+
+    pub fn n_params(&self) -> u32 {
+        self.0.n_params
+    }
+
+    pub fn param_types(&self) -> Vec<Type> {
+        unsafe {
+            let types = self.0.param_types;
+            FromGlibContainerAsVec::from_glib_none_num_as_vec(types, self.n_params() as usize)
+        }
+    }
+}
+
+impl fmt::Debug for SignalQuery {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        f.debug_struct("SignalQuery")
+            .field("signal_name", &self.signal_name())
+            .field("type", &self.type_())
+            .field("flags", &self.flags())
+            .field("return_type", &self.return_type())
+            .field("param_types", &self.param_types())
+            .finish()
+    }
+}
 /// Signal ID.
 #[derive(Debug, Clone, Copy)]
-pub struct SignalId(pub(super) Type, pub(super) u32);
+pub struct SignalId(Type, u32);
+
+impl SignalId {
+    unsafe fn new(type_: Type, id: u32) -> Self {
+        Self(type_, id)
+    }
+
+    /// Find a SignalId by it's `name` and the `type` it connects to
+    #[doc(alias = "g_signal_lookup")]
+    pub fn lookup(name: &str, type_: Type) -> Option<Self> {
+        unsafe {
+            let signal_id = gobject_ffi::g_signal_lookup(name.to_glib_none().0, type_.to_glib());
+            if signal_id == 0 {
+                None
+            } else {
+                Some(Self::new(type_, signal_id))
+            }
+        }
+    }
+
+    pub fn id(&self) -> u32 {
+        self.1
+    }
+
+    pub fn type_(&self) -> Type {
+        self.0
+    }
+
+    /// Queries more in-depth information about the current signal
+    #[doc(alias = "g_signal_query")]
+    pub fn query(&self) -> SignalQuery {
+        unsafe {
+            let query_ptr = std::ptr::null_mut();
```

Comment:
```suggestion
            let mut query_ptr = mem::MaybeUninit::uninit();
```

And `.as_mut_ptr()` and `.assume_init()` below